InvenTree/plugin/installer.py

```diff
@@ -195,30 +236,82 @@ def install_plugin(url=None, packagename=None, user=None):
                 ret['result'] = _(f'Installed plugin into {path}')
 
     except subprocess.CalledProcessError as error:
-        # If an error was thrown, we need to parse the output
+        handle_pip_error(error, 'plugin_install')
 
-        output = error.output.decode('utf-8')
-        logger.exception('Plugin installation failed: %s', str(output))
+    # Save plugin to plugins file
+    add_plugin_to_file(full_pkg)
+
+    # Reload the plugin registry, to discover the new plugin
+    from plugin.registry import registry
+
+    registry.reload_plugins(full_reload=True, force_reload=True, collect=True)
 
-        errors = [_('Plugin installation failed')]
+    return ret
 
-        for msg in output.split('\n'):
-            msg = msg.strip()
 
-            if msg:
-                errors.append(msg)
+def validate_package_plugin(cfg: plugin.models.PluginConfig, user=None):
+    """Validate a package plugin for update or removal."""
+    if not cfg.plugin:
+        raise ValidationError(_('Plugin was not found in registry'))
 
-        if len(errors) > 1:
-            raise ValidationError(errors)
-        else:
-            raise ValidationError(errors[0])
+    if not cfg.is_package():
+        raise ValidationError(_('Plugin is not a packaged plugin'))
 
-    # Save plugin to plugins file
-    add_plugin_to_file(full_pkg)
+    if not cfg.package_name:
+        raise ValidationError(_('Plugin package name not found'))
 
-    # Reload the plugin registry, to discover the new plugin
+    if user and not user.is_staff:
+        raise ValidationError(_('Only staff users can administer plugins'))
+
+
+def uninstall_plugin(
+    cfg: plugin.models.PluginConfig, user=None, delete_config=True, delete_tables=False
+):
+    """Uninstall a plugin from the python virtual environment.
+
+    - The plugin must not be active
+    - The plugin must be a "package" and have a valid package name
+
+    Args:
+        cfg: PluginConfig object
+        user: User performing the uninstall
+        delete_config: If True, delete the plugin configuration from the database
+        delete_tables: If True, delete any database tables associated with the plugin
+    """
     from plugin.registry import registry
 
+    if cfg.active:
+        raise ValidationError(
+            _('Plugin cannot be uninstalled as it is currently active')
+        )
+
+    validate_package_plugin(cfg, user)
+    package_name = cfg.package_name
+    logger.info('Uninstalling plugin: %s', package_name)
+
+    if delete_tables:
+        # TODO: Remove any database tables associated with this plugin
```

Comment:
Question: Will this be part of this PR?

Reply:
Not sure. Don't think so at this stage - this will be a lot trickier.. I will leave it there as a placeholder and make a new issue

Reply:
Removed for now